contracts/migrator/README.md

```diff
@@ -0,0 +1,193 @@
+ * `data: bytes[]`: The ABI-encoding of the `MigrationCallbackData`, to be passed to the Uniswap Liquidity Pool Callback.
+
+#### Function Spec
+
+`function migrate(collateral: (CToken, uint256)[], borrowAmount: uint256) external`
+
+- **REQUIRE** `inMigration == 0`
+- **STORE** `inMigration += 1`
+- **BIND** `user = msg.sender`
+- **WHEN** `repayAmount == type(uint256).max)`:
+  - **BIND READ** `repayAmountActual = cToken.borrowBalanceCurrent(user)`
+- **ELSE**
+  - **BIND** `repayAmountActual = repayAmount`
+- **BIND** `borrowAmountTotal = repayAmountActual + FullMath.mulDivRoundingUp(repayAmountActual, uniswapLiquidityPoolFee, 1e6)` # TODO: FullMath
+- **BIND** `data = abi.encode(MigrationCallbackData{user, repayAmountActual, repayBorrowBehalf, collateral})`
+- **CALL** `uniswapLiquidityPool.flash(address(this), uniswapLiquidityPoolToken0 ? repayAmount : 0, uniswapLiquidityPoolToken0 ? 0 : repayAmount, data)`
+- **STORE** `inMigration -= 1`
+
+Note: for fee calculation see [UniswapV3Pool](https://github.com/Uniswap/v3-core/blob/main/contracts/UniswapV3Pool.sol#L800).
+
+### Uniswap Liquidity Pool Callback Function
+
+This function handles a callback from the Uniswap Liquidity Pool after it has sent this contract the requested tokens. We are responsible for repaying those tokens, with a fee, before we return from this function call.
+
+#### Pre-conditions
+
+This function may only be called during a migration command. We ensure this by making sure this function, itself, is the caller of the `swap` command. We check that the call originates from the expected Uniswap pool, and we check that we are actively processing a migration. This combination of events should ensure that no external party can trigger this code, though it's not clear it would be dangerous even if such a party did.
+
+#### Inputs
+
+ - `uint fee0`: The fee for borrowing token0 from pool. Ignored.
+ - `uint fee1`: The fee for borrowing token1 from pool. Ignored.
+ - `calldata data`: The data encoded above, which is the ABI-encoding of `MigrationCallbackData`.
+
+#### Bindings
+
+ * `repayAmountActual`: The repay amount, after accounting for max.
+
+#### Function Spec
+
+`function uniswapV3FlashCallback(uint256 fee0, uint256 fee1, bytes calldata data)`
+
+  - **REQUIRE** `inMigration == 1`
+  - **REQUIRE** `msg.sender == uniswapLiquidityPool`
+  - **REQUIRE** `sender == address(this)`
+  - **BIND** `MigrationCallbackData{user, repayAmountActual, borrowAmountTotal, collateral} = abi.decode(data, (MigrationCallbackData))`
+  - **CALL** `borrowCToken.repayBorrowBehalf(user, repayAmountActual)`
+  - **FOREACH** `(cToken, amount)` in `collateral`:
+    - **CALL** `cToken.transferFrom(user, amount == type(uint256).max ? cToken.balanceOf(user) : amount)`
+    - **CALL** `cToken.redeem(cToken.balanceOf(address(this)))`
+    - **CALL** `comet.supplyCollateral(address(this), user, cToken.underlying(), cToken.underlying().balanceOf(address(this)))`
```

Comment:
I'm assuming we're calling `balanceOf` here instead of using the `amount` calculated in the `transferFrom` case to handle tokens with transfer fees? I'm not sure if that's totally necessary because `Comet` doesn't support tokens with transfer fees.